devito/nodes.py

```diff
@@ -473,9 +474,19 @@ def __init__(self, name, body, retval, parameters=None, prefix=('static', 'inlin
         self.name = name
         self.body = as_tuple(body)
         self.retval = retval
-        self.parameters = as_tuple(parameters)
         self.prefix = prefix
 
+        if all([isinstance(x, RuntimeArgProvider) for x in parameters]):
+            args = flatten([x.rtargs for x in parameters])
+        else:
+            assert(all([isinstance(x, RuntimeArgument) for x in parameters]))
+            args = parameters
+
+        self.parameters = as_tuple(args)
+
+        self.t_args = [x for x in args if isinstance(x, TensorArgument)]
```

Comment:
There are two good reasons to do it in Devito, and probably two more in general.
For the sake of Devito:
- convention. It's the approach I've used all over the codebase during the 3.0 rewrite
- marking the attribute as read-only. If it's not a property, you could modify if from the outside
In general (ie, might be useful here in the future, or maybe not):
- we may want to do something sophisticated at getting/setting time
- we may have to recompute things on-the-fly because the object state has changed meanwhile